### PR TITLE
feat: Enabling CAST.AI feature | Workload Autoscaler

### PR DESCRIPTION
**Pull request by prasanna-CASTAI**
This PR includes changes to enable a Workload Autoscaler in the module.

Ref: https://docs.cast.ai/docs/workload-autoscaling-configuration

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -180,3 +180,21 @@ variable "wait_for_cluster_ready" {
...
+  description = "Version of castai-workload-autoscaler helm chart. Default latest"
+  type        = string
+  default     = null
+}
+
+variable "workload_autoscaler_values" {
+  description = "List of YAML formatted string with cluster-workload-autoscaler values"
+  type        = list(string)
+  default     = []
+}
```
Comment (aivarasatkocaitiscastai) [inline variables.tf:200]: Would be nice to add these to readme as well

**Review by aivarasatkocaitiscastai — APPROVED** [review]

File: main.tf [hunk 2]
```diff
@@ -656,6 +656,68 @@ resource "helm_release" "castai_kvisor_self_managed" {
...
+  }
+
+  depends_on = [helm_release.castai_agent]
+
+  lifecycle {
+    ignore_changes = [version]
+  }
+}
+
+resource "helm_release" "castai_workload_autoscaler_self_managed" {
```
Comment (varnastadeus) [inline main.tf:693]: why this is needed?

Comment (aivarasatkocaitiscastai) [reply main.tf:693]: same thought but looks like main.tf has all of our components like this..

Comment (prasanna-CASTAI) [reply main.tf:693]: self_managed mode allow users to control the version and deployment of the component directly through Terraform

File: variables.tf (hunk 1 above)
Comment (prasanna-CASTAI) [reply variables.tf:200]: I've updated the README file

File: main.tf (hunk 2 above)
Comment (aivarasatkocaitiscastai) [reply main.tf:693]: @varnastadeus Is it ok to proceed with the PR?

Comment (aldor007) [reply main.tf:693]: @varnastadeus are you fine with merge of this?

Comment (Hy3n4) [reply main.tf:693]: I would also vote for merge and release if it's possible 🙋‍♂️

**Review by varnastadeus — APPROVED** [review]

File: main.tf [hunk 3]
```diff
@@ -656,6 +656,68 @@ resource "helm_release" "castai_kvisor_self_managed" {
...
+    name  = "castai.apiKeySecretRef"
+    value = "castai-cluster-controller"
+  }
+
+  set {
+    name  = "castai.configMapRef"
+    value = "castai-cluster-controller"
+  }
+
+  depends_on = [helm_release.castai_agent, helm_release.castai_workload_autoscaler]
```
Comment (varnastadeus) [inline main.tf:717]:
```suggestion
  depends_on = [helm_release.castai_agent, helm_release.castai_cluster_controller]
```

File: main.tf [hunk 4]
```diff
@@ -656,6 +656,68 @@ resource "helm_release" "castai_kvisor_self_managed" {
...
+    name  = "castai.apiKeySecretRef"
+    value = "castai-cluster-controller"
+  }
+
+  set {
+    name  = "castai.configMapRef"
+    value = "castai-cluster-controller"
+  }
+
+  depends_on = [helm_release.castai_agent]
```
Comment (varnastadeus) [inline main.tf:686]:
```suggestion
  depends_on = [helm_release.castai_agent, helm_release.castai_cluster_controller]
```

**Review by aldor007 — APPROVED** [review]

**Pull request merged**